examples/binarySearch/binarySearch.act

```diff
@@ -0,0 +1,44 @@
+behaviour init of BinarySearch
+interface constructor()
+
+creates
+	uint256[] array := []
+
+invariants
+	sorted(array)
+
+where
+
+sorted : uint256[] -> bool
+sorted([]) = true
+sorted([x]) = true
+sorted(x::y::xs) = x <= y and sorted(y::xs)
+
+
+// Returns the index of elem if elem in array,
+// otherwise returns -1.
+behaviour search of BinarySearch
+interface search(uint256 elem)
+
+returns binarySearch(array, elem, 0, array.length)
+
+ensures
+
+if elem in array
+then array[index] = elem
+else index = -1
+
+where
+
+index = binarySearch(array, elem, 0, array.length)
```

Comment:
```suggestion
index := binarySearch(array, elem, 0, array.length)
```

Reply:
at least that's what we've been doing so far. Maybe it's tedious...

Reply:
but it would leave `=` as comparison, the way you are using it above

Reply:
although we've been using `==` in other examples so far

Reply:
updated `index` and `binarySearch`